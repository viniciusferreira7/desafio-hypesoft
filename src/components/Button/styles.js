import styled, { css } from 'styled-components';

export const Container = styled.button`
${({ theme, background, sizes }) => css`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 1.2rem ${theme.spacings.large};
  letter-spacing: 0.02rem;
  text-transform: uppercase;
  background:${background ? theme.colors.secondaryColor : theme.colors.white};
  color:${background ? theme.colors.white : theme.colors.primaryColor};
  font-size: ${sizes}rem;
  border-radius: 5px;
  border: none;
  transition: all 500ms ease-in-out;
  cursor: pointer;

&:hover {
  background:${!background ? theme.colors.secondaryColor : theme.colors.white};
  color:${!background ? theme.colors.white : theme.colors.primaryColor};
}
`}
`;
