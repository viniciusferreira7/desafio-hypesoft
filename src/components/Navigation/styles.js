import styled, { css } from 'styled-components';

const menuVisible = () => css`
  visibility: visible;
  opacity: 1;
  `;

export const Container = styled.div`
${({ theme, visible }) => css`
    z-index: 5;
    width:100%;
    padding:2.8rem 0 3.8rem;
    border-bottom-color: ${theme.colors.mediumGray};
    transition: all 300ms ease-in-out;

    @media ${theme.media.lteMedium}{
      position: absolute;
      top: 0;
      left: 0;
      height:100vh;
      visibility:hidden;
      opacity: 0;
      background-color:${theme.colors.primaryColor};
      ${visible && menuVisible(theme)}
    }
`}
`;

export const Button = styled.button`
${({ theme, visible }) => css`
  z-index:6;
  position: fixed;
  top: 2rem;
  right:2rem;
  width:4rem;
  height:4rem;
  background: ${
    visible ? theme.colors.secondaryColor : theme.colors.primaryColor
  };
  color:${theme.colors.white};
  border:none;
  display:none;
  pointer-events: ${visible ? 'none' : 'all'};

  > svg{
    width:2.5rem;
    height:2.5rem;
  }

  @media ${theme.media.lteMedium}{
    display:flex;
    align-items: center;
    justify-content: center;
  }
`}`;

export const NavLinks = styled.nav`
${({ theme }) => css`
display:flex;
flex-flow:row wrap;
justify-content: right;

@media ${theme.media.lteMedium}{
  flex-flow: column wrap;
  align-content: center;
}

`}`;

export const Links = styled.a`
${({ theme, visible }) => css`
  display: block;
  font-size: ${theme.font.sizes.small};
  text-transform: uppercase;
  text-decoration: none;
  padding: 0 4.0rem 2rem 4.0rem;
  color:${theme.colors.white};
  position:relative;
  cursor: pointer;

  @media ${theme.media.lteMedium}{
    padding: ${theme.spacings.xlarge} 0;
    font-size: ${theme.font.sizes.medium};
    color:${visible ? theme.colors.primaryColor : theme.colors.white}
  }

  &::after{
    content: '';
    position: absolute;
    bottom: 1.3rem;
    left:50%;
    width:0;
    height:0.2rem;
    background:${theme.colors.secondaryColor};
    transition: all 300ms ease-in-out;
  }

  &:hover::after{
    left:25%;
    width:50%;
  }
  `}
`;
