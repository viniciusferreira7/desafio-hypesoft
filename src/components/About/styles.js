import styled, { css } from 'styled-components';
import { Container as Button } from '../Button/styles';
import { Container as TextContainer } from '../TextComponent/styles';

export const Container = styled.div`
${() => css`
  width: 100%;
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-column-gap: 2.8rem;
  grid-template-areas:'SrcImg' 'Description';
  padding-top: 10rem;
`}
`;

export const SrcImg = styled.div`
${({ theme, srcImg }) => css`
  grid-area: 'SrcImg';
  max-width: 56rem;
  width: 100%;
  height: 54.2rem;
  background-image: url(${srcImg});
  background-size: cover;
  border-radius: 1rem;
  box-shadow: 3.2rem 3.2rem   ${theme.colors.white};

  @media ${theme.media.lteMedium}{
    display: none;
  }
`}
`;

export const Description = styled.div`
${() => css`
  grid-area: 'Description';
  max-width: 67.2rem;

  > ${Button} {
    width: 16.2rem;
    height: 5.6rem;

    > img{
      margin-left: 1.1rem;
    }
  }
`}
`;

export const InfoHeading = styled.div`
${() => css`
`}
`;

export const InfoContainer = styled.div`
${({ theme }) => css`
  max-width: 67.2rem;
  display: flex;

  @media ${theme.media.lteMedium}{
    flex-direction: column;
    padding-bottom: 3.2rem;
  }
`}
`;

export const Group = styled.div`
${() => css`
  max-width: 20.6rem;
  display: flex;
  align-items: center;
  margin-right: 5rem;

  > ${TextContainer}:nth-child(2){
    max-width: 9.5rem;
    margin-left: 1.6rem;
    padding: 0;
  }
`}
`;

`}`;

export const GroupTwo = styled.div`
${() => css`
  width: 25.1rem;
  display: flex;
  align-items: center;

  > ${TextContainer}:nth-child(2){
    max-width: 9.5rem;
    margin-left: 1.6rem;
    padding: 0;
  }
`}
`;

`}`;
