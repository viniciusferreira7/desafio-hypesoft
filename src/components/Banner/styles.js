import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';
import { Container as Button } from '../Button/styles';
import { Container as GridDot } from '../GridDot/styles';

export const Container = styled.section`
${({ theme }) => css`
  width: 100%;
  max-height:31.2rem;
  overflow: hidden;
  padding: 6.4rem 2%;
  position: relative;
  margin-top: 16.4rem;

  @media ${theme.media.lteMedium}{
      max-height:auto;
      padding-top: 3rem
    }

 >  ${GridDot}{
    opacity: 0.15;
    position: absolute;
    top: 6.4rem;
    left: 12.2rem;
  }


  ${Heading}{
      width:50rem;
      height: 9.6rem;
      margin: 0 auto;


      @media ${theme.media.lteMedium}{
        width: 100%;
        text-align: center;
        height: 15rem;
        margin-bottom: 7rem;
      }

      > span{
        color:#bc199c;
      }

      > span::before{
        display: none;
      }
    }

  > ${Button} {
      width: 17.7rem;
      height: 5.6rem;
      margin: 0 auto;
      margin-top: 7.2rem;

      > a{
        margin-right: 0.9rem;
      }
  }
`}
`;

export const SmallEllipse = styled.div`
  ${({ theme }) => css`
    width:7.6rem;
    height:7.6rem;
    border-radius: calc(7.6rem / 2);
    background: ${theme.colors.white};
    opacity: 0.03;
    position: absolute;
    top: 2.6rem;
    right: 11.6rem;

    @media ${theme.media.lteMedium}{
      top: 8rem;
      right: 0;
    }
  `}
`;

export const HugeEllipse = styled.div`
  ${({ theme }) => css`
    width:52.9rem;
    height:52.9rem;
    border-radius: calc(52.9rem / 2);
    background: ${theme.colors.white};
    opacity: 0.03;
    position: absolute;
    top: 13.9rem;
    right: 90.7rem;

    @media ${theme.media.lteMedium}{
      width:30rem;
      height:30rem;
      top: 0;
      right: 4rem;
    }
  `}
`;

export const ContainerGridDot = styled.div`
  ${() => css`
    position: absolute;
    top:15rem;
    right: 21rem;
    transform: rotate(-90deg);

    ${GridDot}{
      opacity: 0.3;

    }
  `}
`;
