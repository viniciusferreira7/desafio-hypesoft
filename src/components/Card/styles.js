import styled, { css } from 'styled-components';
import { Container as TextContainer } from '../TextComponent/styles';

export const Container = styled.div`
${({ theme }) => css`
  background: linear-gradient(0deg, rgba(223, 227, 235, 0.4), rgba(223, 227, 235, 0.4)), #FFFFFF;
  border-radius: 6px;
  width: 31.2rem;
  max-height: 35.1rem;
  transition: all 400ms ease-in-out;

  @media ${theme.media.lteMedium}{
    width: 18.5rem;
    padding-bottom: 2%;
    max-height: 40rem;

    > svg {
      width:2.5rem !important;
      height:2.5rem !important;
      margin: 0;
    }

    ${TextContainer}{
      text-align: center;
      font-size: 2rem;
      padding:2.2rem 0 3.2rem 0 !important ;

      > strong {
        margin-top: 1.6rem;
      }
    }

  }

    > svg{
      width:4rem;
      height:4rem;
      margin: 2.2rem 2.2rem 0 2.2rem;
      color: #242424;
    }

    ${TextContainer}{
      padding:2.2rem 0 3.2rem 2.8rem ;
      position: relative;

      > strong {
        width: 6.4rem;
        height: 0.3rem;
        background-color: #ED1CA6;
        margin-top: 1.6rem;
        margin-bottom: 2.8rem;
        position: absolute;
        bottom: -1.3rem;
        left:6rem ;
      }
    }

    > img{
      margin: 2.8rem 0 3.6rem 3.5rem;
    }


    &:hover{
      background: #FFFFFF;
      box-shadow: 0px 20px 48px rgba(10, 25, 49, 0.09);
      border-radius: 6px;
      border: 0.3rem solid #ED1CA6;

      ${TextContainer}{
        color:#ED1CA6 ;
      }

      > svg {
        color: #ED1CA6;
      }
    }

;
`}
`;
