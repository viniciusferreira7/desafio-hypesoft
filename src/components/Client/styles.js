import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
${() => css`
  display: flex;
  align-items: center;
  width: 31.2rem;
  height: 10rem;
  background: linear-gradient(0deg, rgba(223, 227, 235, 0.3), rgba(223, 227, 235, 0.3)), #FFFFFF;
  border-radius: 0.4rem;
`}
`;

export const Photo = styled.div`
${({ theme, srcImg }) => css`
  width: 5.2rem;
  height: 5.2rem;
  border-radius: calc(5.2rem / 2);
  background-image:url(${srcImg});
  background-size: cover;
  background-color: ${theme.colors.gray_500};
  margin: 2.4rem;
  `}
`;

export const Info = styled.div`
${() => css`
  ${TextComponent}{
    text-transform: capitalize;
    padding: 0;
    padding-bottom: 0.6rem;

    > span{
      color: #ED1CA6
;
    }
  }
`}
`;
