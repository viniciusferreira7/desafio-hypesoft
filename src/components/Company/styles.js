import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';

export const Container = styled.div`
${({ theme }) => css`
  width: 100%;
  display: flex;
  padding-top: 20rem;
  justify-content: space-around;

  @media ${theme.media.lteMedium}{
    flex-direction: column;
    text-align: center;
  }

  ${Heading}{
    width: 312px;

    > strong{
      color:#bc199c;
      font-weight: 400;
    }
  }
`}
`;

export const ContainerLogo = styled.div`
${({ theme }) => css`
  display: flex;
  max-width: 93.6rem;
  max-height: 12.0rem;
  justify-content: space-around;

  @media ${theme.media.lteMedium}{
    flex-direction: column;
    width: 100%;
    max-height: 96.6rem;
  }
`}
`;
export const Logo = styled.div`
${({ theme }) => css`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 168px;
  height: 120px;
  margin-right: 2.4rem;
  background: linear-gradient(0deg, rgba(223, 227, 235, 0.3), rgba(223, 227, 235, 0.3)), #FFFFFF;
  border-radius: 0.8rem;
  transition: all 500ms ease-in-out;

  @media ${theme.media.lteMedium}{
    margin: 0 auto;
    margin-top: 3rem;
  }

  &:hover{
    border:1rem solid ${theme.colors.white};
  }

  >img{
    width: 11.304rem;
    height: 2.6rem;
  }

  >img:nth-of-type(5){
    width:2.756rem !important;
    height:1.205rem !important;
  }

;

`}
`;
