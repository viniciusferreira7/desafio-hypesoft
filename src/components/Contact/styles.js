import styled, { css } from 'styled-components';
import { Container as GridDot } from '../GridDot/styles';

export const Container = styled.section`
${({ theme }) => css`
  width: 100%;
  display: flex;
  padding: 10rem;
  align-items: center;
  margin-top: 22.7rem;
  position: relative;

  ${GridDot}{
    position: absolute;
    top: 16%;
    right:12%;
    transform: rotate(-90deg);
    opacity: 0.3;

    @media ${theme.media.lteMedium}{
      display: none;
  }
  }

  @media ${theme.media.lteMedium}{
    padding: 2rem;
    flex-direction: column;
  }
`}`;
