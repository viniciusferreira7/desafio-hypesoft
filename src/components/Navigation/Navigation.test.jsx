import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { Navigation } from '.';
import { fireEvent } from '@storybook/testing-library';
import { theme } from '../../styles/theme';

const mock = {
  links_1: 'link 1',
  links_2: 'link 2',
  links_3: 'link 3',
  links_4: 'link 4',
};

describe('<Navigation/>', () => {
  it('should render links', () => {
    renderTheme(<Navigation {...mock} />);
    expect(screen.getByText(/link 1/i)).toBeInTheDocument();
    expect(screen.getByText(/link 2/i)).toBeInTheDocument();
    expect(screen.getByText(/link 3/i)).toBeInTheDocument();
    expect(screen.getByText(/link 4/i)).toBeInTheDocument();
  });

  it('should render links', () => {
    renderTheme(<Navigation {...mock} />);
    expect(screen.getByText(/link 4/i).parentElement).toHaveStyleRule(
      'flex-flow',
      'column wrap',
      {
        media: theme.media.lteMedium,
      },
    );
  });

  it('should render menu mobile an button for open and close the menu', () => {
    renderTheme(<Navigation {...mock} />);

    const button = screen.getByLabelText('Open/Close menu');
    const menuContainer = button.nextSibling;

    expect(button).toHaveStyleRule('display', 'none');
    expect(button).toHaveStyleRule('display', 'flex', {
      media: theme.media.lteMedium,
    });
    expect(menuContainer).toHaveStyleRule('opacity', '0', {
      media: theme.media.lteMedium,
    });

    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();

    fireEvent.click(button);

    expect(menuContainer).toHaveStyleRule('opacity', '1', {
      media: theme.media.lteMedium,
    });
    expect(screen.getByLabelText('Close menu')).toBeInTheDocument();

    fireEvent.click(menuContainer);

    expect(menuContainer).toHaveStyleRule('opacity', '0', {
      media: theme.media.lteMedium,
    });

    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<Navigation {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
