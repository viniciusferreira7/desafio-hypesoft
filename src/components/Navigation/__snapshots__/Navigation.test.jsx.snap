// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation/> should match snapshot 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c2 {
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  z-index: 5;
  width: 100%;
  padding: 2.8rem 0 3.8rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0 {
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 2rem;
  width: 4rem;
  height: 4rem;
  background: #242424;
  color: #ffff;
  border: none;
  display: none;
  pointer-events: all;
}

.c0 > svg {
  width: 2.5rem;
  height: 2.5rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: right;
  -webkit-justify-content: right;
  -ms-flex-pack: right;
  justify-content: right;
}

.c4 {
  display: block;
  font-size: 1.6rem;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 0 4.0rem 2rem 4.0rem;
  color: #242424;
  position: relative;
}

.c4::after {
  content: '';
  position: absolute;
  bottom: 1.3rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #7D00FF;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c4:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width:768px) {
  .c2 {
    height: 100vh;
    visibility: hidden;
    opacity: 0;
    background-color: #242424;
  }
}

@media (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:768px) {
  .c3 {
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
    -webkit-align-content: center;
    -ms-flex-line-pack: center;
    align-content: center;
  }
}

@media (max-width:768px) {
  .c4 {
    color: #ffff;
  }
}

<div>
  <button
    aria-label="Open/Close menu"
    class="c0"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="c2"
  >
    <nav
      class="c3"
    >
      <a
        class="c4"
      >
        link 1
      </a>
      <a
        class="c4"
      >
        link 2
      </a>
      <a
        class="c4"
      >
        link 3
      </a>
      <a
        class="c4"
      >
        link 4
      </a>
    </nav>
  </div>
</div>
`;
